### PR TITLE
New package: Enlightenment 16 (e16)

### PR DESCRIPTION
**Pull request by IntinteDAO**
Enlightenment version 16 is an old (but developed) version of the Enlightenment environment. What makes it different from, for example, OpenBox is its very cool look, graphical effects and automation (e.g. menu generation).

In my opinion, it's a very cool environment to promote Termux, and functional at the same time. Better, of course, would still be Enlightenment version 25, however, it is much more complicated.

### CODE REVIEW

**Review by TomJo2000 — APPROVED** [review]
This PR gets a tentative LGTM from me.

I would like to urge you to verify that everything that should be in the package is in the package and verify functionality on-device.

I'm also somewhat miffed that there doesn't appear to be an `enlightenment-16` branch nor tag on the upstream repo, e.g [the Enlightenment Gitea instance](https://git.enlightenment.org/enlightenment/enlightenment/).
Would have been nicer to use a tarball from that instead of sourceforge.

**Comment by IntinteDAO** [comment]
Thank you. I would also prefer it to be from gitea or another source, but I guess they only keep it on Sourceforge.

The best would be to port the newer version of Enlightenment (this one is also being developed, but it's like KDE3 and KDE5), but the level of complexity is much higher. For now "for fun". I'm trying to do it, but "the game is hard".

It would be nice if you could somehow enable AUTO UPDATE, but I have never used this functionality and I don't know if it would work and if it would work correctly.

**Comment by Biswa96** [comment]
I have tested the aarch64 package from CI artifact in my phone. It's working fine with lxqt.

**Comment by IntinteDAO** [comment]
> I have tested the aarch64 package from CI artifact in my phone. It's working fine with lxqt.

Well, E16 can operate independently. I don't know if you need to use LXQT for this. This is the way I tested it.
On the other hand, it is good to have another opinion.

File: x11-packages/e16/0001-fix-autogenerate-menu.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff -ur orig/scripts/e_gen_menu mod/scripts/e_gen_menu
+--- orig/scripts/e_gen_menu	2024-06-24 12:38:21.000000000 +0200
++++ mod/scripts/e_gen_menu	2024-10-28 16:56:02.340591608 +0100
+@@ -362,7 +362,7 @@
+         dbg += 1
+ 
+ # Likely  prefixes
+-Prefixes  = ['/usr/local', '/usr', '/opt']
++Prefixes  = ['/usr/local', '/usr', '/opt', '@TERMUX_PREFIX@/usr']
```
Comment (truboxl) [inline x11-packages/e16/0001-fix-autogenerate-menu.patch:9]:
```suggestion
+Prefixes  = ['/usr/local', '/usr', '/opt', '@TERMUX_PREFIX@']
```

File: x11-packages/e16/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+TERMUX_PKG_HOMEPAGE=https://www.enlightenment.org/e16
+TERMUX_PKG_DESCRIPTION='E16 (Enlightenment) is a lightweight, highly customizable window manager for X11, known for its unique aesthetic, efficiency, and low resource usage, making it ideal for older or limited-resource systems.'
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=1.0.30
+TERMUX_PKG_SRCURL=https://downloads.sourceforge.net/enlightenment/e16-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=b8b7748a2c48c4c7c9758d9ad12b14f566d2bec38f2eda533e6d874f5ce9074c
+TERMUX_PKG_DEPENDS="imlib2"
```
Comment (truboxl) [inline x11-packages/e16/build.sh:8]:
```suggestion
TERMUX_PKG_DEPENDS="glib, imlib2, libiconv, libx11, libxext, libxft, libxrender, pango"
```

**Comment by Biswa96** [comment]
> Well, E16 can operate independently. I don't know if you need to use LXQT for this

I have tested without lxqt also but I can not open any application in e16 directly (there is not taskbar/panel). I have to open another terminal tab to run an application.

**Comment by IntinteDAO** [comment]
> > Well, E16 can operate independently. I don't know if you need to use LXQT for this
> 
> I have tested without lxqt also but I can not open any application in e16 directly (there is not taskbar/panel). I have to open another terminal tab to run an application.

In theory, you should have a menu with applications under the left mouse button. If you don't have it, it may be a bug in the menu generator (I corrected it and it works for me).

I will still see the truboxl fix

**Comment by IntinteDAO** [comment]
> > Well, E16 can operate independently. I don't know if you need to use LXQT for this
> 
> I have tested without lxqt also but I can not open any application in e16 directly (there is not taskbar/panel). I have to open another terminal tab to run an application.

That's the idea behind the desktop. In the sense, except for the clock (and probably tray icon / indicator) everything works, at least initially.

For now, I'm working on E25, it should be better for users

**Review by truboxl — APPROVED** [review]

File: x11-packages/e16/build.sh (hunk 2 above)
Comment (TomJo2000) [reply x11-packages/e16/build.sh]: Usually you can use project documentation and other build scripts for the same package (https://repology.org/projects/) as a guide.
`ldd` can also be helpful, though that only shows what the executable or shared object links to,
so you'd need to use it on a compiled version of the package.

Comment (truboxl) [reply x11-packages/e16/build.sh]: I crafted a bash function that uses readelf and apt-file for identifying dependencies on device to review PRs.
I am waiting #21847 or looking into whether its possible to do it on CI...

**Pull request merged**